tests/realtikvtest/addindexlittest/workload.go

```diff
@@ -0,0 +1,258 @@
+package addindexlittest
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pingcap/tidb/testkit"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	skippedErrorCode = []string{"8028", "9007"}
+	wCtx             = workloadContext{}
+)
+
+func InitWorkLoadParameters(ctx *suiteContext) {
+	ctx.HasWorkload = true
+	ctx.EndWorkload = false
+	ctx.Wl = newWorkloadContext()
+	ctx.tkPool = &sync.Pool{New: func() interface{} {
+		return testkit.NewTestKit(ctx.t, ctx.store)
+	}}
+}
+
+func newWorkloadContext() workload {
+	return &wCtx
+}
+
+func initWorkLoadContext(ctx *suiteContext, ID ...int) {
+	if len(ID) < 2 {
+		return
+	}
+
+	wCtx.tableID = ID[0]
+	wCtx.tableName = "t" + strconv.Itoa(wCtx.tableID)
+	wCtx.colID = wCtx.colID[:0]
+	for i := 1; i < len(ID); i++ {
+		wCtx.colID = append(wCtx.colID, ID[i])
+	}
+	// set started insert id to 10000
+	wCtx.insertID = 10000
+	wCtx.date = "2008-02-02"
+}
+
+type workChan struct {
+	err      error
+	finished bool
+}
+
+type worker struct {
+	id       int
+	loadType int // 0 is insert, 1 update, 2 delete
+	wChan    chan *workChan
+}
+
+type workloadContext struct {
+	tableName string
+	tableID   int
+	colID     []int
+	workerNum int
+	wr        []*worker
+	insertID  int
+	date      string
+}
+
+func newWorker(wId int, ty int) *worker {
+	wr := worker{
+		id:       wId,
+		loadType: ty,
+		wChan:    make(chan *workChan, 1),
+	}
+	return &wr
+}
+
+func (w *worker) run(ctx *suiteContext) {
+	var err error
+	tk := ctx.getTestKit()
+	tk.MustExec("use addindexlit")
+	for {
+		if ctx.EndWorkload {
+			break
+		}
+		if w.loadType == 0 {
+			// insert workload
+			err = insertWorker(ctx, tk)
+		} else if w.loadType == 1 {
+			// update workload
+			err = updateWorker(ctx, tk)
+		} else if w.loadType == 1 {
```

Comment:
```suggestion
		} else if w.loadType == 2 {
```

Reply:
Let's keep it to reproduce the issue until we solve it.